src/IIIF/IIIF/ImageApi/ImageRequest.cs

```diff
@@ -52,6 +57,11 @@ public static ImageRequest Parse(string path, string prefix)
             request.IsInformationRequest = true;
             return request;
         }
+        
+        if (validateSegments && (parts.Length != 5 || parts.Any(string.IsNullOrEmpty)))
+        {
+            throw new ArgumentException("Path contains empty or an invalid number of parameters");
```

Comment:
Is it better using "segments" rather than "parameters" in this error message - keeps consistent with `validateSegments` arg?

`"Path contains empty or an invalid number of segments"`